### PR TITLE
Refactor Results::Success#filter_issues to mutate its attribute

### PR DESCRIPTION
**Pull request by yykamei**
See https://github.com/sider/runners/pull/655#discussion_r364537300

### CODE REVIEW

File: lib/runners/harness.rb [hunk 1]
```diff
@@ -49,8 +49,9 @@ def run
               case result
               when Results::Success
                 trace_writer.message "Removing issues from unchanged or untracked files..." do
-                  result.filter_issues(changes).tap do |new_result|
-                    new_result.add_git_blame_info(workspace)
+                  result.tap do |r|
+                    r.filter_issues(changes)
+                    r.add_git_blame_info(workspace)
```
Comment (ybiquitous) [inline lib/runners/harness.rb]: `tap` is needed?

```ruby
result.filter_issues(changes)
result.add_git_blame_info(workspace)
```

Comment (ybiquitous) [reply lib/runners/harness.rb]: Sorry, it's needed.

File: lib/runners/results.rb [hunk 2]
```diff
@@ -62,11 +62,9 @@ def add_issue(issue)
       end
 
       def filter_issues(changes)
-        result = self.class.new(guid: guid, analyzer: analyzer)
-        issues.each do |issue|
-          result.add_issue(issue) if changes.include?(issue)
+        issues.reject! do |issue|
```
Comment (ybiquitous) [inline lib/runners/results.rb]: How about this? See http://ruby-doc.org/core-2.7.0/Array.html#//dash_ref_method%252Di%252Dfilter%252D21/Method/filter!/0

```suggestion
        issues.filter! do |issue|
```

Comment (yykamei) [reply lib/runners/results.rb]: Oh, I mistakenly used `reject!`! I should have chosen `filter!`. Thank you!

Comment (yykamei) [reply lib/runners/results.rb]: I replaced `reject!` with `filter!` and did git-commit(1) with `--amend`.

Comment (yykamei) [reply lib/runners/results.rb]: Sorry, reject! is correct. I'm looking into the error on CI.

Comment (yykamei) [reply lib/runners/results.rb]: `filer!` is correct...

File: sig/polyfill.rbi [hunk 3]
```diff
@@ -60,6 +60,7 @@ extension Array (Polyfill)
   def +: <'x> (Array<'x>) -> Array<'a | 'x>
        | (self) -> self
   def filter: { ('a) -> bool } -> self
+  def filter!: { ('a) -> bool } -> void
```
Comment (yykamei) [inline sig/polyfill.rbi]: `Array#filter!` returns `nil` or its instance, but we should not depend on the response of this method, so I set the return type `void`.

https://ruby-doc.org/core-2.7.0/Array.html#method-i-filter-21

**Comment by ybiquitous** [comment]
Why...? 

> NoMethodError: undefined method `filter!' for #<Array:0xXXXXXX>

**Comment by ybiquitous** [comment]
`#Array:0xXXXXXX` does not seem an `Array`... 🤔

**Comment by yykamei** [comment]
I understood. `Array#filter!` does not exist in Ruby 2.5. I'll use `Array#select! instead.

File: sig/polyfill.rbi (hunk 3 above)
Comment (ybiquitous) [reply sig/polyfill.rbi]: Is this unneeded?

Comment (yykamei) [reply sig/polyfill.rbi]: Yes. Thank you

**Pull request merged**